subiquity/ui/views/filesystem/delete.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2018 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+from urwid import Text
+
+from subiquitycore.ui.buttons import danger_btn, other_btn
+from subiquitycore.ui.utils import button_pile
+from subiquitycore.ui.stretchy import Stretchy
+
+
+log = logging.getLogger('subiquity.ui.filesystem.disk_info')
+
+
+class ConfirmDeleteStretchy(Stretchy):
```

Comment:
I saw this in the asciinima; since these are constructed items, not real partitions, do we want to impose the confirmation dialog?

Reply:
I don't know. I'll talk to design I guess. The ideal thing would probably be to not have the dialog but have a way of undoing a mistaken delete. No idea at all how to implement that though!

I was going to include in the dialog a description about the objects that would be implicitly deleted by the deletion (so in the case of a partition that could be filesystem and mount) but apparently I didn't get to that :) I should probably wait until we've decided if this dialog is a good idea before implementing that...